common/src/main/java/io/github/notenoughupdates/moulconfig/gui/editors/GuiOptionEditorButton.java

```diff
@@ -50,7 +55,9 @@ public GuiOptionEditorButton(
         this.config = config;
 
         this.buttonText = buttonText;
-        this.isUsingRunnable = option.getType() == Runnable.class;
+        Type type = option.getType();
+        this.isUsingRunnable = type == Runnable.class;
+        this.isUsingKotlinLambda = type instanceof ParameterizedType && ((ParameterizedType) type).getRawType() == KFunction.class;
         if (this.buttonText == null) this.buttonText = "";
```

Comment:
out of scope: we should probably add the option to check if a runnableId is valid with the config: i.e. insert a call to `config.validateRunnable(runnableId)` and `Warnings.warn` if it isnt. (with a default implementation of just `return true` for compat)